lib/db/structs.go

```diff
@@ -75,15 +79,15 @@ func (f FileInfoTruncated) SequenceNo() int64 {
 	return f.Sequence
 }
 
-func (f FileInfoTruncated) ConvertToInvalidFileInfo(invalidatedBy protocol.ShortID) protocol.FileInfo {
+func (f FileInfoTruncated) ConvertToIgnoredFileInfo(by protocol.ShortID) protocol.FileInfo {
 	return protocol.FileInfo{
 		Name:         f.Name,
 		Type:         f.Type,
 		ModifiedS:    f.ModifiedS,
 		ModifiedNs:   f.ModifiedNs,
-		ModifiedBy:   invalidatedBy,
-		Invalid:      true,
+		ModifiedBy:   by,
 		Version:      f.Version,
 		RawBlockSize: f.RawBlockSize,
+		LocalFlags:   protocol.FlagLocalIgnored,
```

Comment:
I've been thinking about that. At the moment, with the flags as currently defined, it does not make sense to set more than one at a time (it won't be both ignored and must-rescan for example). It could equally well be an enum, really. I think this is the cleaner behavior.

The same goes for the SetIgnored, SetMustRescan and SetUnsupported methods in bep_extensions

Reply:
There is currently no precedence as they are mutually exclusive.

An `ignored` file is not scanned so it does not become `unsupported` (unless it's the other way around, I don't remember) and isn't pulled so can never get the `must rescan` flag.

A file with the `must rescan` flag will have its flag cleared on the next scan no matter what. It might possibly become `ignored` at that point if the ignore patterns changed at the same time.

We could make the setters do `OR`, but then we would also need another method to clear the other flags that we always use before using the setter since that's always what we want at the moment.

It could possibly matter once receive only folders materalize, as a local change that is also ignored might then want both `ignored` and `receive only changed` flags... Let me think about that since I'm roughly at that stage now anyway.

Reply:
What you say makes sense yet I'd just have some branchy switch statement ensuring that we don't break the state transitions we believe in.

Reply:
Right, so I was beginning to agree with you that the reasonable behavior for flags is to OR in a method called `SetWhateverFlag`, but then I realized that that's not what these methods are. They also set versions and clear block lists and stuff, so they are utility methods to accomplish a FileInfo for a given purpose - a file that is ignored and has no blocks and other flags, for example. If and when we need to do flag twiddling only it's something else.